packages/menu-bar/src/vaadin-menu-bar.d.ts

```diff
@@ -77,20 +79,26 @@ export interface MenuBarEventMap extends HTMLElementEventMap, MenuBarCustomEvent
  *
  * @fires {CustomEvent} item-selected - Fired when a submenu item or menu bar button without children is clicked.
  */
-declare class MenuBar extends MenuBarMixin(DisabledMixin(ElementMixin(ThemableMixin(HTMLElement)))) {
-  addEventListener<K extends keyof MenuBarEventMap>(
+declare class MenuBar<TItem extends MenuBarItem = MenuBarItem> extends HTMLElement {
+  addEventListener<K extends keyof MenuBarEventMap<TItem>>(
     type: K,
-    listener: (this: MenuBar, ev: MenuBarEventMap[K]) => void,
+    listener: (this: MenuBar, ev: MenuBarEventMap<TItem>[K]) => void,
```

Comment:
Should we use `this: MenuBar<TItem>` here? Other components with generics seem to do that.

Reply:
Done